libs/fetcher.js

```diff
@@ -273,8 +259,8 @@ function executeRequest(request, resolve, reject) {
  *      the req object, the serviceInfo object, and the params object.  It is expected to return the processed params object.
  * @constructor
  */
-function Fetcher(options) {
-    this.options = options || {};
+function Fetcher(options = {}) {
```

Comment:
should we just make this a class now?

Reply:
It can be. I didn't change at first because of the size of the diff. But since you pointed, I converted to modern class.